admin/views.py

```diff
@@ -871,8 +886,34 @@ def update_approval_status():
         sale_receipt = SaleReceiptModel.query.get(receipt_id)
         if action == 'approve':
             sale_receipt.is_approved = True
+            sale_receipt.token_amount = sale_receipt.promised_token
+
+            # Add earning wallet plan to user membership start
+            membership_plan_id = ""
+
+            if sale_receipt.token_amount == 10:
+                membership_plan_id = "earned_days_01"
+            elif sale_receipt.token_amount == 8:
+                membership_plan_id = "earned_days_02"
+            elif sale_receipt.token_amount == 6:
+                membership_plan_id = "earned_days_03"
+            elif sale_receipt.token_amount == 2:
+                membership_plan_id = "earned_days_04"
+
+            user_membership = UserMembership(
+                user_id=sale_receipt.user_id,
+                membership_plan_id=membership_plan_id,
+                payment_src_id="earned_days",
+                notes="earned via sale receipt"
+            )
+            db.session.add(user_membership)
+            db.session.commit()
+            # Add earning wallet plan to user membership end
+
+            return jsonify({'success': True})
```

Comment:
these lines are not needed as there's a db.session.commit() and return statement already mentioned outside the if/else condition. So better to not change the return format we have and avoid repeating same code.

Reply:
done.